index.js

```diff
@@ -48,6 +72,44 @@ module.exports = {
     }
 
     await Promise.all(tasks);
+
+    if (this.isExistingMonorepo) {
+      await this.moveFilesForExistingMonorepo(options);
+
+      this.ui.writeWarnLine(
+        `Make sure your workspaces are configured correctly to cover the newly created ${addonInfo.location} and ${testAppInfo.location} packages!`
+      );
+    }
+  },
+
+  // EmberCLI will always create a <addon-name> root folder for us, which makes no sense in an existing repo
+  // Until we have a better solution for this use case upstream, we are fixing this here by moving the generated
+  // files outside of this folder and deleting it
+  async moveFilesForExistingMonorepo(options) {
+    let addonInfo = addonInfoFromOptions(options);
+    let testAppInfo = testAppInfoFromOptions(options);
+
+    let tmpDir = await createTmp();
+    let tmpAddonDir = path.join(tmpDir, 'addon');
+    let tmpTestAppDir = path.join(tmpDir, 'test-app');
+    let originalAddonDir = path.resolve(addonInfo.location);
+    let originalTestAppDir = path.resolve(testAppInfo.location);
+    let finalAddonDir = path.resolve('..', addonInfo.location);
+    let finalTestAppDir = path.resolve('..', testAppInfo.location);
+    let extraneousDir = process.cwd();
+
+    await fs.move(originalAddonDir, tmpAddonDir);
+    await fs.move(originalTestAppDir, tmpTestAppDir);
+
+    for (let fileToCopy of filesToCopyFromRootToAddon) {
+      await fs.move(fileToCopy, path.join(tmpAddonDir, fileToCopy));
+    }
+
+    process.chdir('..');
+    await fs.remove(extraneousDir);
```

Comment:
what's the liklihood of the extraneousDir existing as some other folder in a monorepo? would ember-cli then error about the folder not being empty?

Reply:
> would ember-cli then error about the folder not being empty?

Yes, it will! Just double-checked.